ethpm_cli/commands/install.py

```diff
@@ -119,6 +129,93 @@ def get_package_aliases(package_name: str, config: Config) -> Iterable[Tuple[str
                 yield alias
 
 
+def resolve_pkg_name_or_alias(pkg_name: str, config: Config) -> InstalledPackage:
+    lockfile_path = config.ethpm_dir / "ethpm.lock"
+    lockfile = json.loads(lockfile_path.read_text())
+    return InstalledPackage(**lockfile[pkg_name])
+
+
+def update_package(args: Namespace, config: Config) -> None:
+    if is_package_installed(args.package, config):
+        installed_pkg = resolve_pkg_name_or_alias(args.package, config)
+        active_registry = get_active_registry(config.xdg_ethpmcli_root / REGISTRY_STORE)
+        if is_valid_registry_uri(installed_pkg.install_uri):
+            validate_same_registry(installed_pkg.install_uri, active_registry.uri)
+
+        connected_chain_id = config.w3.eth.chainId
+        active_registry_uri = parse_registry_uri(active_registry.uri)
+        if not to_int(text=active_registry_uri.chain_id) == connected_chain_id:
+            raise InstallError(
+                f"Registry URI chain: {active_registry_uri.chain_id} doesn't match "
+                f"connected web3: {connected_chain_id}."
+            )
+
+        config.w3.pm.set_registry(active_registry_uri.address)
+        all_pkg_names = config.w3.pm.get_all_package_names()
+        if installed_pkg.resolved_package_name not in all_pkg_names:
+            raise InstallError(
+                f"{installed_pkg.resolved_package_name} is not available on the active registry "
+                f"{active_registry.uri}. Available packages include: {all_pkg_names}."
+            )
+
+        all_release_data = config.w3.pm.get_all_package_releases(
+            installed_pkg.resolved_package_name
+        )
+        all_versions = [version for version, _ in all_release_data]
+
+        if installed_pkg.resolved_version not in all_versions:
+            raise InstallError(
+                f"{installed_pkg.resolved_package_name}@{installed_pkg.resolved_version} not "
+                f"found on the active registry {active_registry.uri}."
+            )
+
+        on_chain_install_uri = pluck_release_data(
+            all_release_data, installed_pkg.resolved_version
+        )
+        if on_chain_install_uri != installed_pkg.resolved_uri:
+            raise InstallError(
+                f"Install URI found on active registry for {installed_pkg.resolved_package_name}@"
+                f"{installed_pkg.resolved_version}: {on_chain_install_uri} does not match the "
+                f"install URI found in local lockfile: {installed_pkg.resolved_uri}."
+            )
+
+        cli_logger.info(
+            f"{len(all_versions)} versions of {installed_pkg.resolved_package_name} "
+            f"found on the active registry: {all_versions}."
+        )
+        while True:
```

Comment:
I'd put an upper bound on the number of iterations here and you can error out a final `else` clause.  Generally good to avoid infinite loops.